### ISSUE TITLE
kube-register gets stuck sometimes, does not register minions

### ISSUE THREAD

**Issue by LuqmanSahaf** [kelseyhightower/kube-register#14]
Hi,

`kube-register` has been creating problems for few days. I will first explain how I setup my cluster and then the problem.

### Cluster Setup
I setup cluster of `Kubernetes` on `CoreOS` on DigitalOcean. I use `systemd` units in `cloud-config`(s) to download Kubernetes and other binaries and run them as service on three (or more) nodes.

### Problem
The problem is that `kube-register` sometimes gets stuck and does not work. Other times, it just works fine. Whenever, it gets stuck it does not register any minions in the cluster.
When I restart the `kube-register` service it starts working and successfully registers all the minions.

Following are some logs and debugging I had done:
#### Logs

The logs from the `kube-register` program are:

```
Mar 30 17:06:45 test-master systemd[1]: Started Kubernetes Registration Service.
Mar 30 17:06:45 test-master kube-register[910]: error health checking 10.132.58.196: Get http://10.132.58.196:10250/healthz: dial tcp 10.132.58.196:10250: connection refused
Mar 30 17:06:45 test-master kube-register[910]: error health checking 10.132.153.74: Get http://10.132.153.74:10250/healthz: dial tcp 10.132.153.74:10250: connection refused
```

The logs do not show significant info.

#### `top` output
Interestingly, the output of `top` command shows that `kube-register` isn't using any resource, CPU nor memory. I monitored it for more than a minute.

```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                   
  910 root      20   0    9044   4236   3672 S   0.0  0.0   0:00.15 kube-register
```

#### `netstat` output
The `netstat` output for `kube-register` gives that kube-register has an established connection with the a minion `10.132.229.147`.

```
sudo netstat -anp | grep kube-register
tcp        0      0 10.132.58.196:42809     10.132.229.147:10250    ESTABLISHED 910/kube-register   
unix  3      [ ]         STREAM     CONNECTED     24735    910/kube-register
```

Please, help me in solving this issue as it's creating hurdle in automation of `Kubernetes` cluster formation. Thanks.

**Comment by LuqmanSahaf** [comment]
@vmarmol

**Comment by vmarmol** [comment]
@LuqmanSahaf I now see #13 is it possibly the same issue? What version of Kubernetes are you running?

**Comment by LuqmanSahaf** [comment]
@vmarmol I'm using 0.12.1. I think it's different because in #13 there is API change that's causing `kube-register` to break. Moreover, as mentioned in the post `kube-register` does work but sometimes it gets stuck and does not do anything.

**Comment by LuqmanSahaf** [comment]
It turns out that the problem is not with the `kube-register`. I added more logs into kube-register to debug the issue. I found out that the `kube-register` gets stuck at [this line](https://github.com/kelseyhightower/kube-register/blob/master/machines.go#L50). It tries to check the health of minions. The minions do not reply. The minions were stuck and there is no issue with `kube-register`.
I think the problem was with the orchestration of minions. When minions start before master, they get stuck.

I am closing this issue.
